torch/_functorch/aot_autograd.py

```diff
@@ -1702,6 +1702,18 @@ def functionalized_f_helper(*args):
                     assert inpt_new is not inpt_old
                     inpt_old.copy_(inpt_new)
 
+        if aot_config.keep_inference_input_mutations and trace_joint:
+            for i, (inpt_old, inpt_f) in enumerate(zip(args[0], f_args[0])):
```

Comment:
args[1] will be inputs to backward so we ignore them here

Reply:
Instead of these two c onditions:
```
if aot_config.keep_inference_input_mutations and not trace_joint:
    ...
if aot_config.keep_inference_input_mutations and trace_joint:
    ...
```

Can you try to merge them together? (I think you'll just have to do e.g. `args_, f_args_ = args[0], f_args[0] if trace_joint else args, f_args`)